xapian-letor/include/xapian-letor/ranker.h

```diff
@@ -193,7 +195,7 @@ class XAPIAN_VISIBILITY_DEFAULT Ranker : public Xapian::Internal::intrusive_base
   protected:
     /// Method to train the model. Overridden in ranker subclass.
     virtual void
-    train(const std::vector<Xapian::FeatureVector> & training_data) = 0;
+    train(const vector<vector<Xapian::FeatureVector>> & training_data) = 0;
```

Comment:
Yes, that's right - we're now aiming to follow the modern C++ convention of putting `*` and `&` with the type (so `const vector<vector<Xapian::FeatureVector>>&` here) with a space between that and a variable/function name that follows.

A lot of the existing code doesn't follow this, as it predates us adopting this, but please follow it for new and changed code so at least the situation doesn't get worse.  At some point we'll do an automated fix of the remaining cases.

(We used to follow typical C-style which is to write `char *p` or `char * p`, and used the obvious extension to reference types, which C itself doesn't have).

The patch checking script will now spot many cases of this, but maybe not this one (it needs to avoid flagging when the `&` is a bit-wise and operation as that should have a space either side).

Reply:
I just checked and it is reported by the current version of the script:

```
xapian-letor/include/xapian-letor/ranker.h:196: error: Preferred style is 'int& x' (not 'int &x' or 'int & x'): +    train(const vector<vector<Xapian::FeatureVector>> & training_data) = 0;
```

Reply:
fixed in a fixup commit.